src/context/product-context.js

```diff
@@ -0,0 +1,19 @@
+import { createContext, useContext, useReducer } from "react";
+import {productReducer} from "../reducer/product-reducer";
+
+const ProductContext = createContext();
+
+const ProductProvider = ({children}) => {
+
+    const [state,dispatch] = useReducer(productReducer , {products: [] , sortByPrice: "" , sortByRating: "" , sortByCategory: {mobile: false , laptop: false} , priceRange: 50000 })
+
+    return(
+        <ProductContext.Provider value={{state,dispatch}}>
```

Comment:
here instead of sending the whole state, you can destructure this and then pass.

Reply:
will look into it